src/Resources/public/reference-entity/record/table/value/value-updater.ts

```diff
@@ -34,6 +37,32 @@ export default class ValueUpdater {
         }
     }
 
+    public createDragOver(): (event: React.DragEvent<HTMLTableRowElement>) => void {
+        return (event: React.DragEvent<HTMLTableRowElement>) => {
+            event.preventDefault();
+        };
+    }
+
+    public createDragStart(index: number): (event: React.DragEvent<HTMLTableRowElement>) => void {
+        return () => {
+            this.dragPosition = index;
+        };
+    }
+
+    public createDrop(index: number): (event: React.DragEvent<HTMLTableRowElement>) => void {
+        return (event: React.DragEvent<HTMLTableRowElement>) => {
+            event.preventDefault();
+
+            const row: TableDataRow | undefined = this.internalDataRows.splice(this.dragPosition, 1).shift();
```

Comment:
- `splice` and `shift` mutate the array, so here be aware
- i guess would be better to define types as `TableDataRow | null`, `null` indicates that the row does not exist

Reply:
> splice and shift mutate the array, so here be aware

Good point. Can you please provide a test where this might result into a problem with the current code to see if we already have an issue with the state if the array?

> i guess would be better to define types as TableDataRow | null, null indicates that the row does not exist

The return type of `shift()` is typed as `TableDataRow | undefined` here. Not having `undefined` breaks the analysis build.

Reply:
> The return type of shift() is typed as TableDataRow | undefined here. Not having undefined breaks the analysis build.

nice 👍

> Good point. Can you please provide a test where this might result into a problem with the current code to see if we already have an issue with the state if the array?

see the answer on line 62